skyportal/handlers/api/image_analysis.py

```diff
@@ -0,0 +1,451 @@
+            mask=mask,
+            fwhm=fwhm,
+            gain=gain,
+            aper=1.0,
+            bkgann=[5, 7],
+            sn=5,
+            verbose=True,
+        )
+        log(f'{len(obj)} objects properly measured')
+
+        # ## Astrometric calibration
+        # Getting the center position, size and pixel scale for the image
+        center_ra, center_dec, center_sr = astrometry.get_frame_center(
+            header=header, width=image.shape[1], height=image.shape[0]
+        )
+        pixscale = astrometry.get_pixscale(header=header)
+
+        log(
+            f'Frame center is {center_ra} {center_dec} radius {center_sr} deg, {pixscale*3600} arcsec/pixel'
+        )
+
+        # ## Reference catalogue
+        # Catalogue name may be any Vizier identifier (ps1, gaiadr2, gaiaedr3, usnob1, gsc, skymapper, apass, sdss, atlas, vsx).
+        # Getting PanSTARRS objects brighter than r=17 mag
+        cat = catalogs.get_cat_vizier(
+            center_ra, center_dec, center_sr, 'ps1', filters={'rmag': '<21'}
+        )
+        log(f'{len(cat)} catalogue stars')
+
+        # ## Astrometric refinement
+        # Refining the astrometric solution based on the positions of detected objects and catalogue stars with scamp
+        wcs = pipeline.refine_astrometry(
+            obj, cat, wcs=wcs, method='scamp', cat_col_mag='rmag', verbose=True
+        )
+        if wcs is not None:
+            # Update WCS info in the header
+            astrometry.clear_wcs(
+                header,
+                remove_comments=True,
+                remove_underscored=True,
+                remove_history=True,
+            )
+            header.update(wcs.to_header(relax=True))
+
+        # ## Photometric calibration
+
+        # Positionally matching detected objects with catalogue stars
+        # Then building the photometric model for their instrumental magnitudes
+
+        # Photometric calibration using 2 arcsec matching radius, r magnitude, g-r color and second order spatial variations
+        m = pipeline.calibrate_photometry(
+            obj,
+            cat,
+            sr=2 / 3600,
+            cat_col_mag='rmag',
+            cat_col_mag1='gmag',
+            cat_col_mag2='rmag',
+            max_intrinsic_rms=0.02,
+            order=2,
+            verbose=True,
+        )
+
+        # The code above automatically augments the object list with calibrated magnitudes, but we may also do it manually
+        # obj['mag_calib'] = obj['mag'] + m['zero_fn'](obj['x'], obj['y'])
+        # obj['mag_calib_err'] = np.hypot(obj['magerr'], m['zero_fn'](obj['x'], obj['y'], get_err=True))
+
+        # ## Simple catalogue-based transient detection
+        # Some transients may already be detected by comparing the detected objects with catalogue BUT limited approach
+
+        # Filtering of transient candidates
+        candidates = pipeline.filter_transient_candidates(
+            obj, cat=cat, sr=2 / 3600, verbose=True
+        )
```

Comment:
I don't know if `verbose=True` is what you'd want in production.

Reply:
Same reply as above @guynir42 . I would like to keep this during the review/changes process and remove it later on if that's ok with you of course.